plugins/inputs/netflow/README.md

```diff
@@ -0,0 +1,101 @@
+# Netflow Input Plugin
+
+The `netflow` plugin acts as a collector for Netflow v5, Netflow v9 and IPFIX
+flow information. The Layer 4 protocol numbers are gathered from the
+[official IANA assignments][IANA assignments].
+The internal field mappings for Netflow v5 fields are defined according to
+[Cisco's Netflow v5 documentation][CISCO NF5], Netflow v9 fields are defined
+according to [Cisco's Netflow v9 documentation][CISCO NF9] and the
+[ASA extensions][].
+Definitions for IPFIX are according to [IANA assignement document][IPFIX doc].
+
+[IANA assignments]: https://www.iana.org/assignments/protocol-numbers/protocol-numbers.xhtml
+[CISCO NF5]:        https://www.cisco.com/c/en/us/td/docs/net_mgmt/netflow_collection_engine/3-6/user/guide/format.html#wp1006186
+[CISCO NF9]:        https://www.cisco.com/en/US/technologies/tk648/tk362/technologies_white_paper09186a00800a3db9.html
+[ASA extensions]:   https://www.cisco.com/c/en/us/td/docs/security/asa/special/netflow/asa_netflow.html
+[IPFIX doc]:        https://www.iana.org/assignments/ipfix/ipfix.xhtml#ipfix-nat-type
+
+## Configuration
+
+```toml @sample.conf
+# Netflow v5, Netflow v9 and IPFIX collector
+[[inputs.netflow]]
+  ## Address to listen for netflow/ipfix packets.
+  ##   example: service_address = "udp://:2055"
+  ##            service_address = "udp4://:2055"
+  ##            service_address = "udp6://:2055"
+  service_address = "udp://:2055"
+
+  ## Set the size of the operating system's receive buffer.
+  ##   example: read_buffer_size = "64KiB"
+  # read_buffer_size = ""
+
+  ## Protocol version to use for decoding.
+  ## Available options are
+  ##   "netflow v5" -- Netflow v5 protocol
+  ##   "netflow v9" -- Netflow v9 protocol (also works for IPFIX)
+  ##   "ipfix"      -- IPFIX / Netflow v10 protocol (also works for Netflow v9)
+  # protocol = "ipfix"
+
+  ## Dump incoming packets to the log
+  ## This can be helpful to debug parsing issues. Only active if
+  ## Telegraf is in debug mode.
+  # dump_packets = false
```

Comment:
Any boolean is going to be scrutinized :) would we ever add any further debug logging that would not be dumping packets?

Reply:
By default we don't want this enables as it will print out the raw, binary UDP packet as byte-array. I do have a small tool that can convert this back to the actual message to quickly spin-up unit-tests with that data, but it might fill your screen. So it is useful if you trip over "unknown" messages or messages triggering bugs but no for everyday use (please note we do have some people running with debug-enabled in production).

Reply:
Right so I'm ok with the option and that it is disabled by default, my concern is the naming and future possibility of adding additional debugging that would require more boolean flags.

Reply:
Right now I cannot think of any additional flags for debugging. The only remotely imaginable thing is to trace the decoding, but that does not exist in the used library and it would be one more flag at maximum.

For the naming I am open for suggestions! Maybe `debug_log_raw_packets`?

Reply:
let's keep it then :)